### ISSUE TITLE
com.google.gapid.server.Client$InternalServerErrorException: Tracing Failed

### ISSUE THREAD

**Issue by pemgithub** [google/gapid#2835]
I'm able to take a GLES trace of a debug build of gles3jni on my rooted Android device.

However, when I try to take a GLES trace of 3DMark (or any games in the google play store), I get the following error message.  My device is rooted, so I think it should work, but instead I get the error message...  Android Studio is not open.

com.google.gapid.server.Client$InternalServerErrorException: Tracing Failed
   Cause: Could not start trace
   Cause: Unable to connect to the application.

This can happen when another debugger or IDE is running in the background, such as Android Studio.
Please close any running Android debuggers and try again.

See https://github.com/google/gapid/issues/911 for more information
	at com.google.gapid.server.Client.throwIfError(Client.java:301)
	at com.google.gapid.server.Client.lambda$streamTrace$49(Client.java:275)
	at com.google.gapid.server.GapidClientGrpc$StreamHandler.onNext(GapidClientGrpc.java:217)
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onMessage(ClientCalls.java:421)
	at io.grpc.ForwardingClientCallListener.onMessage(ForwardingClientCallListener.java:33)
	at io.grpc.ForwardingClientCallListener.onMessage(ForwardingClientCallListener.java:33)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1MessagesAvailable.runInContext(ClientCallImpl.java:519)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.google.gapid.server.Client$Stack: For request: RPC->streamTrace()
	at com.google.gapid.server.Client.streamTrace(Client.java:272)
	at com.google.gapid.server.Tracer.trace(Tracer.java:41)
	at com.google.gapid.views.TracerDialog.showTracingDialog(TracerDialog.java:132)
	at com.google.gapid.MainWindow.lambda$updateFileMenu$11(MainWindow.java:255)
	at com.google.gapid.MainWindow$MenuItems$1.run(MainWindow.java:399)
	at org.eclipse.jface.action.Action.runWithEvent(Action.java:476)
	at org.eclipse.jface.action.ActionContributionItem.handleWidgetSelection(ActionContributionItem.java:568)
	at org.eclipse.jface.action.ActionContributionItem.lambda$4(ActionContributionItem.java:400)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4145)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1055)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:3958)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3561)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:821)
	at org.eclipse.jface.window.Window.open(Window.java:797)
	at com.google.gapid.Main$UI.show(Main.java:119)
	at com.google.gapid.Main.main(Main.java:68)

**Comment by AWoloszyn** [comment]
This can happen for a number of reasons that I can think of.
1) Something else is trying to attach to the jdwp port instead.
2) The applications in question are failing for other reasons. For example, if the application forks or runs in a different process, this will fail as well. Or they detect they are being debugged and close.
3) Something else that we don't know about yet.

Does tracing something simpler like Calculator work for you?

**Comment by pemgithub** [comment]
Tracing com.google.android.calculator fails with the same error message.

Another idea... I was surprised to that GAPID app (com.google.android.gapid.aarch64 or com.google.android.gapid.armabi) does not request root from Magisk Manager.  For Root Checker Basic and Shell, the apps attempt root, Magisk Manager detects it, then I can enable root per app in Magisk Manager app > Superuser.

**Comment by pemgithub** [comment]
GAPID apparently does not support trace recording of non-debuggable APK (such as Play Store app).  Even on a rooted Pixel 3a XL.

I used apktool to add android:debuggable="true" to com.learnopengles.android, and I'm able to trace it.

However, when I tried this with actual games (eg Clash of Clans), it doesn't work.  I think it's because modifying the APK requires a new signature, and I guess Clash of Clans (and probably most Play Store games) verify expected signature.

https://gapid.dev/trace/ says "Either a debuggable application, or a device running a ‘rooted’ user-debug build."...  My Pixel 3a XL is using https://developers.google.com/android/images -> "bonito" for Pixel 3a XL, which was rooted using Magisk.  Does that qualify as rooted user-debug?
